### PR TITLE
[SPARK-23085][ML] API parity for mllib.linalg.Vectors.sparse 

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?
`ML.Vectors#sparse(size: Int, elements: Seq[(Int, Double)])` support zero-length

## How was this patch tested?
existing tests

### CODE REVIEW

**Comment by mgaido91** [comment]
as far as I know mllib is not maintained anymore, thus I am not sure this fix is useful

**Comment by srowen** [comment]
That's true, although, there's no conceptual reason not to support a 0-size vector, and the existing checks even checked for >= 0 in one case. This is arguably just fixing internal consistency. How about a unit test?

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Unless someone like @jkbradley or @MLnick objects, I think this doesn't hurt and actually makes things a little more consistent.

File: mllib/src/test/scala/org/apache/spark/mllib/linalg/VectorsSuite.scala [hunk 1]
```diff
@@ -113,6 +113,13 @@ class VectorsSuite extends SparkFunSuite with Logging {
     assert(vec.toArray === arr)
   }
 
+  test("zero-length sparse vector") {
```
Comment (MLnick) [inline mllib/src/test/scala/org/apache/spark/mllib/linalg/VectorsSuite.scala]: We may as well add the same test to `ml.linalg`

Comment (MLnick) [inline mllib/src/test/scala/org/apache/spark/mllib/linalg/VectorsSuite.scala]: While we're doing this we may as well also add a test to `intercept` the exception for negative size (as per the other sparse vector construction tests), for both `ml` and `mllib`

**Review by MLnick — CHANGES_REQUESTED** [review]
Seems fine to add for consistency, even though `mllib` is no longer actively developed.

But we should round out the tests - I left a couple comments on that.

**Review by MLnick — APPROVED** [review]
LGTM

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**